client/service-container-chain/src/spawner.rs

```diff
@@ -256,16 +256,19 @@ async fn try_spawn<SelectSyncMode: TSelectSyncMode>(
     if !start_collation {
```

Comment:
Add something like this as per previous comment

```suggestion
    if !start_collation && not_data_preserver {
```

Reply:
*fixed*